tidy3d/components/simulation.py

```diff
@@ -242,6 +242,58 @@ def bloch_with_symmetry(cls, val, values):
                 )
         return val
 
+    # pylint: disable=too-many-locals
+    @pydantic.validator("boundary_spec", always=True)
+    def plane_wave_boundaries(cls, val, values):
+        """Error if there are plane wave sources incompatible with boundary conditions."""
+        boundaries = val.to_list
+        sources = values.get("sources")
+        size = values.get("size")
+        sim_medium = values.get("medium")
+        structures = values.get("structures")
+        for src_idx, source in enumerate(sources):
+            if not isinstance(source, PlaneWave):
+                continue
+
+            _, tan_dirs = cls.pop_axis([0, 1, 2], axis=source.injection_axis)
+            medium_set = cls.intersecting_media(source, structures)
+            medium = medium_set.pop() if medium_set else sim_medium
+
+            for tan_dir in tan_dirs:
+                boundary = boundaries[tan_dir]
+
+                # check the PML/absorber + angled plane wave case
+                num_pml = sum(isinstance(bnd, (PML, StablePML, Absorber)) for bnd in boundary)
+                if num_pml > 0 and source.angle_theta != 0:
+                    raise SetupError(
+                        "Angled plane wave sources are not compatible with the PML boundary along "
```

Comment:
Actually sorry, let's just call it absorbing boundary. My previous comment was to say "absorbing or PML", but I think we can just go with absorbing.